### PR TITLE
Add wolfSSL_EC_KEY_get_conv_form to compatibility layer.

### PR DESCRIPTION
**Pull request by haydenroche5**
# Description

See title.

# Testing

Added `test_wolfSSL_EC_KEY_set_conv_form`, which tests the getter and setter.

# Checklist

 - [x] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -40244,6 +40244,14 @@ void wolfSSL_EC_KEY_set_conv_form(WOLFSSL_EC_KEY *eckey, char form)
     }
 }
 
+point_conversion_form_t wolfSSL_EC_KEY_get_conv_form(const WOLFSSL_EC_KEY* key)
+{
+    if (key != NULL) {
+        return key->form;
+    }
+
+    return -1;
```
Comment (SparkiDev) [inline src/ssl.c:40253]: Test to confirm NULL check.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

**Comment by haydenroche5** [comment]
The customer wanting this is using FIPS. I'll need to find a way to do this without modifying ecc.h/ecc.c.

File: tests/api.c [hunk 2]
```diff
@@ -44847,6 +44847,34 @@ static void test_wolfSSL_EC_KEY_set_group(void)
...
 
+static void test_wolfSSL_EC_KEY_get_conv_form(void)
+{
+#if defined(HAVE_ECC) && defined(OPENSSL_EXTRA)
+    BIO* bio;
+    EC_KEY* key = NULL;
+
+    printf(testingFmt, "test_wolfSSL_EC_KEY_set_conv_form");
+
+    AssertIntLT(EC_KEY_get_conv_form(key), 0);
```
Comment (SparkiDev) [inline tests/api.c:44858]: Add test where NULL is passed to EC_KEY_get_conv_form().

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: tests/api.c (hunk 2 above)
Comment (haydenroche5) [reply tests/api.c:44858]: `key` is NULL here.

Comment (SparkiDev) [reply tests/api.c:44858]: Make it explicit. (Was fooled by the AssertIntLT instead of expected AssertIntEq.)

**Review by SparkiDev — APPROVED** [review]

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**